lms/djangoapps/edxnotes/helpers.py

```diff
@@ -0,0 +1,349 @@
+        return None
+
+    return json.dumps(preprocess_collection(user, course, collection), cls=NoteJSONEncoder)
+
+
+def get_endpoint(path=""):
+    """
+    Returns endpoint.
+    """
+    try:
+        url = settings.EDXNOTES_INTERFACE['url']
+        if not url.endswith("/"):
+            url += "/"
+
+        if path:
+            if path.startswith("/"):
+                path = path.lstrip("/")
+            if not path.endswith("/"):
+                path += "/"
+
+        return url + path
+    except (AttributeError, KeyError):
+        raise ImproperlyConfigured(_("No endpoint was provided for EdxNotes."))
+
+
+def get_course_position(course_module):
+    """
+    Return the user's current place in the course.
+
+    If this is the user's first time, leads to COURSE/CHAPTER/SECTION.
+    If this isn't the users's first time, leads to COURSE/CHAPTER.
+
+    If there is no current position in the course or chapter, then selects
+    the first child.
+    """
+    urlargs = {'course_id': course_module.id.to_deprecated_string()}
+    chapter = get_current_child(course_module, min_depth=1)
+    if chapter is None:
+        log.debug("No chapter found when loading current position in course")
+        return None
+
+    urlargs['chapter'] = chapter.url_name
+    if course_module.position is not None:
+        return {
+            'display_name': chapter.display_name_with_default,
+            'url': reverse('courseware_chapter', kwargs=urlargs),
+        }
+
+    # Relying on default of returning first child
+    section = get_current_child(chapter, min_depth=1)
+    if section is None:
+        log.debug("No section found when loading current position in course")
+        return None
+
+    urlargs['section'] = section.url_name
+    return {
+        'display_name': section.display_name_with_default,
+        'url': reverse('courseware_section', kwargs=urlargs)
+    }
+
+
+def generate_uid():
+    """
+    Generates unique id.
+    """
+    return uuid4().int  # pylint: disable=no-member
+
+
+def is_feature_enabled(course):
+    """
+    Returns True if Student Notes feature is enabled for the course,
+    False otherwise.
+
+    In order for the application to be enabled it must be:
+        1) enabled globally via FEATURES.
+        2) present in the course tab configuration.
+        3) Harvard Annotation Tool must be disabled for the course.
+    """
+    tab_found = next((True for t in course.tabs if t["type"] == "edxnotes"), False)
+    feature_enabled = settings.FEATURES.get("ENABLE_EDXNOTES")
+
+    return (feature_enabled and tab_found) and not is_harvard_notes_enabled(course)
+
+
+def is_harvard_notes_enabled(course):
+    """
+    Returns True if Harvard Annotation Tool is enabled for the course,
+    False otherwise.
+
+    Checks for 'textannotation', 'imageannotation', 'videoannotation' in the list
+    of advanced modules of the course.
+    """
+    modules = set(['textannotation', 'imageannotation', 'videoannotation'])
```

Comment:
> What about the older "annotatable"?

The main idea to avoid using different versions of `AnnotatorJS` with their own changes at the same time.
`annotatable` module doesn't use `AnnotatorJS`, that's why it's not in this list and we can safely use it.